backend/src/index.ts

```diff
@@ -1,64 +1,59 @@
 const MONGODB_URI = `mongodb://${process.env.MONGODB_USER}:${process.env.MONGODB_PASSWORD}@ac-yztvzc4-shard-00-00.b5tvnqi.mongodb.net:27017,ac-yztvzc4-shard-00-01.b5tvnqi.mongodb.net:27017,ac-yztvzc4-shard-00-02.b5tvnqi.mongodb.net:27017/?ssl=true&replicaSet=atlas-dbcw9j-shard-0&authSource=admin&retryWrites=true&w=majority`;
 const startConnection = async () => {
-  try {
-    const dbName = 'contacts';
-
-    await mongoose.connect(MONGODB_URI, { dbName });
-
-    //Create indexes and shard the collection
-    // await Contact.createIndexes();
-    // mongoose.connection.db.admin().command({ shardCollection: `${dbName}.Contact`, key: { clientId: 1 } });
-
-    const port = process.env.PORT || 3000;
-    app.listen(port);
-
-    console.log(`server running at ${process.env.URL}:${port}`);
-  } catch (err) {
+try {
+  const dbName = 'contacts';
+  await mongoose.connect(MONGODB_URI, { dbName });
+  const port = process.env.PORT || 3000;
+  app.listen(port);
+  console.log(`server running at ${process.env.URL}:${port}`);
+} catch (err) {
     console.log(err);
   }
 };
 startConnection();
 
-// Enable CORS for specific origins with credentials
 const corsOptions = {
   origin: `${process.env.FRONTEND_URL}:${process.env.FRONTEND_PORT}`, // Replace with your frontend URL
-  credentials: true,
 };
 
-//order important needs to come before express.json()
-app.use(cors(corsOptions));
+//cors order important: needs to come before express.json()
+// app.use(cors(corsOptions));
 app.use(cookieParser());
-app.use(express.urlencoded({ extended: false })); //handling <form> post data, "false" - parsing the URL-encoded data with the querystring library or the qs library (when true)
-app.use(express.json()); //parse json application/json (replaces need for body-parser)
+app.use(express.urlencoded({ extended: false }));  //handling <form> post data, "false" - parsing the URL-encoded data with the querystring library or the qs library (when true)
+app.use(express.json());  //parse json application/json
 app.use(express.json({ type: 'application/vnd.api+json' })); // Middleware to parse incoming JSON data with JSON API content type
+//static file handling
+app.use(express.static(path.join(__dirname, '../', 'public')));
+app.use('/images', express.static(path.join(__dirname, '../', 'images'))); //serving files as if from root folder but we need /images says we are looking in /images on root
 
 app.use('/contacts', contactRoutes);
 app.use('/auth', authRoutes);
 
+//handle all misc routes
 app.use((req, res) => {
   res.status(404).json({ status: 'ERROR', message: 'Page Not Found' });
 });
 
 // catches all errors passed with next(err);
-app.use((err: ErrorWithStatus, req: Request, res: Response, next: NextFunction) => {
-  console.log(err);
-  const status = err.statusCode || 500;
-  const message = err.message;
-  res.status(status).json({ message });
+//note you need "next:NextFunction" prop as its part of the middleware error handler function prop signature
```

Comment:
I would suggest introducing the convention `_` for unused parameters.

Reply:
done